File: shared/javascript/src/interfaces/IServerData.js
goog.provide("pebble.shared.interfaces.IServerData");


//-----file start
/**
 * @interface
 */
pebble.shared.interfaces.IServerData = function() {};

/**
 * @param {string} collection
 * @param {pebble.Pebble} pebble
 * @return {pebble.Pebble}
 */
pebble.shared.interfaces.IServerData.prototype.create = function (collection, pebble){};
/**
 * @param {string} collection
 * @param {string} uniqueName
 * @return {pebble.Pebble}
 */
pebble.shared.interfaces.IServerData.prototype.retrieve = function(collection, uniqueName){};
/**
 * @param {string} collection
 * @param {string} uniqueName
 * @param {pebble.Pebble} doc
 */
pebble.shared.interfaces.IServerData.prototype.update = function (collection, uniqueName, doc){};
/**
 * @param {string} collection
 * @param {string} uniqueName
 */
pebble.shared.interfaces.IServerData.prototype.deleteItem = function(collection, uniqueName){};
/**
 * @param {Array.<string>} list
 */
pebble.shared.interfaces.IServerData.prototype.deleteCollections = function (list){};
/**
 * @param {pebble.Pebble} queryItem
 * @return {Array.<string>} 
 */
pebble.shared.interfaces.IServerData.prototype.doQuery = function(queryItem){};

File: shared/csharp/src/interfaces/IServerData.cs
using System.Collections.Generic;
using PebbleFields.PebbleObject;

namespace PebbleFields.Shared.Interfaces
{
	/// <summary>
	/// 
	/// </summary>
	/**
	 * The ServerData is where the data is stored
	 * -caching is done here and should be reset on delete and update
	 * @author Ryan Tsunoda
	 *
	 */
	public interface IServerData {

		Pebble create(string collection, Pebble pebble);
		Pebble retrieve(string collection, string uniqueName);
		void update(string collection, string uniqueName, Pebble doc);
		void delete(string collection, string uniqueName);
		void deleteCollections(List<string> list);
		List<string> doQuery(Pebble queryItem);

	}
}
